src/pages/billing/Project.js

```diff
@@ -226,61 +230,50 @@ const ProjectDetail = ({ authorizeAndLoadAccounts, billingAccounts, billingProje
     { namespace: billingProject.projectName },
     _.map('workspace', workspaces)
   ), [billingProject, workspaces])
-
-  const maxWorkspacesInChart = 10
   const spendChartOptions = {
     chart: { type: 'bar', style: { fontFamily: 'inherit' } },
     credits: { enabled: false },
-    legend: { enabled: false },
-    series: [{ name: 'Total Cost', data: costPerWorkspace.workspaceCosts }],
+    legend: { reversed: true },
+    plotOptions: { series: { stacking: 'normal' } },
+    series: [
+      { name: 'Compute', data: costPerWorkspace.computeCosts },
+      { name: 'Storage', data: costPerWorkspace.storageCosts },
+      { name: 'Other', data: costPerWorkspace.otherCosts }
+    ],
     title: {
       text: costPerWorkspace.numWorkspaces > maxWorkspacesInChart ? `Top ${maxWorkspacesInChart} Spending Workspaces` : 'Spend By Workspace'
     },
-    tooltip: { valuePrefix: '$' },
+    tooltip: {
+      followPointer: true,
+      shared: true,
+      headerFormat: '{point.key}',
+      pointFormatter: function() { // eslint-disable-line object-shorthand
+        return `<br/><span style="color:${this.color}">\u25CF</span> ${this.series.name}: ${costPerWorkspace.costFormatter.format(this.y)}`
+      }
+    },
     xAxis: {
       categories: costPerWorkspace.workspaceNames, crosshair: true,
       labels: { style: { fontSize: '12px' } }
     },
     yAxis: {
       crosshair: true, min: 0,
-      labels: { format: `\${value:.2f}`, style: { fontSize: '12px' } },
-      title: { text: 'Total Cost' },
+      labels: {
+        formatter: function() { return costPerWorkspace.costFormatter.format(this.value) }, // eslint-disable-line object-shorthand
+        style: { fontSize: '12px' }
+      },
+      title: { enabled: false },
       width: '96%'
     },
+    accessibility: {
+      point: {
+        descriptionFormatter: point => {
+          return `${point.index + 1}. Workspace ${point.category}, ${point.series.name}: ${costPerWorkspace.costFormatter.format(point.y)}.`
```

Comment:
Overriding for costFormatter, added "Workspace" to make the screen reader text more understandable.